src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -189,21 +178,18 @@ public static Index loadIndex(final String indexFile, Function<SeekableByteChann
     public static Index loadIndex(final String source, final InputStream inputStream) {
         // Must be buffered, because getIndexType uses mark and reset
         try (BufferedInputStream bufferedInputStream = new BufferedInputStream(inputStream, Defaults.NON_ZERO_BUFFER_SIZE)) {
-            final Class<Index> indexClass = IndexType.getIndexType(bufferedInputStream).getIndexType();
-            final Constructor<Index> ctor = indexClass.getConstructor(InputStream.class);
-            return ctor.newInstance(bufferedInputStream);
-        } catch (final TribbleException ex) {
-            throw ex;
-        } catch (final InvocationTargetException ex) {
-            if (ex.getCause() instanceof EOFException) {
+            return createIndex(bufferedInputStream);
+        } catch (final EOFException ex) {
                 throw new TribbleException.CorruptedIndexFile("Index file is corrupted", source, ex);
-            }
-            throw new RuntimeException(ex);
-        } catch (final Exception ex) {
-            throw new RuntimeException(ex);
+        } catch (final IOException ex) {
+            throw new RuntimeIOException("Failed to read index file: " + source, ex);
```

Comment:
Is there a reason why this is `RuntimeException` not `TribbleException`?

Reply:
Historical weirdness.  Could be a TribbleException instead.